pkg/helpers/upgrade.go

```diff
@@ -0,0 +1,88 @@
+package helpers
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/pior/dad/pkg/config"
+	"github.com/pior/dad/pkg/executor"
+)
+
+type Upgrade struct {
+	github    *Github
+	plateform string
+	client    *http.Client
+	skipSudo  bool
+}
+
+const dadPrefix string = "dad-"
+
+func NewUpgrade(cfg *config.Config) (u *Upgrade) {
+	return NewUpgradeWithHTTPClient(cfg, http.DefaultClient)
+}
+
+func NewUpgradeWithHTTPClient(cfg *config.Config, client *http.Client) (u *Upgrade) {
+	g := NewGithubWithClient(cfg, client)
+	plateform := cfg.Platform()
+
+	return &Upgrade{
+		github:    g,
+		plateform: plateform,
+		client:    client,
+	}
+}
+
+func (u *Upgrade) Perform(target string, release *GithubReleaseItem) (err error) {
```

Comment:
`target` is a bit too generic, `destinationPath` ?

Reply:
Do you think we could pass just a `url string` instead of a `* GithubReleaseItem`?
(in order to limit the coupling)